pandas/core/construction.py

```diff
@@ -549,3 +549,38 @@ def _try_cast(
         else:
             subarr = np.array(arr, dtype=object, copy=copy)
     return subarr
+
+
+# see gh-17261
+def is_empty_data(data):
+    """
+    Utility to check if a Series is instantiated with empty data
+    """
+    is_none = data is None
+    is_simple_empty = isinstance(data, (list, tuple, dict)) and not data
```

Comment:
In principle, this could also be an empty generator or other collections, so I think we need to make this check more general. 
Maybe use `is_list_like` but then check it has no `dtype` (for array, series, etc)